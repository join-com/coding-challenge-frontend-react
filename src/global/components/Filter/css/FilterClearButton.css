.FilterClearButton {
  background-color: var(--filter-clear-background-color);
  margin-left: var(--section-margin);
}

.FilterClearButton__text {
  color: var(--filter-clear-color);
}
