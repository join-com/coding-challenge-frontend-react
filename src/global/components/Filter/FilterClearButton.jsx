import React from 'react';
import PropTypes from 'prop-types';

import I18n from '../../../utils/I18n';

import Button from '../Button';

import './css/FilterClearButton.css';

const FilterClearButton = ({ className, handleClearFilters }) => (
  <Button
    className={`FilterClearButton ${className}`}
    text={I18n.t('en.filter.clearFilters')}
    iconName="remove"
    onClick={handleClearFilters}
  />
);

export default FilterClearButton;

FilterClearButton.propTypes = {
  className: PropTypes.string,
  handleClearFilters: PropTypes.func,
};

FilterClearButton.defaultProps = {
  className: '',
  handleClearFilters: () => {},
};
