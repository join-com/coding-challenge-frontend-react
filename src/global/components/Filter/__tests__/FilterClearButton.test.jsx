import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import FilterClearButton from '../FilterClearButton';

const props = {
  className: 'MyFilterClearButtonClassname',
  handleClearFilters: () => {},
};

describe('FilterClearButton', () => {
  it('should render the component correctly', () => {
    const tree = shallow(<FilterClearButton {...props} />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });
});
