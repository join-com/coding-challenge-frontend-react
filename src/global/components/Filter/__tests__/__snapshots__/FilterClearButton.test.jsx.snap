// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterClearButton should render the component correctly 1`] = `
<ButtonWrapper
  className="FilterClearButton MyFilterClearButtonClassname"
  iconName="remove"
  onClick={[Function]}
  params={Object {}}
  text="Clear Filters"
/>
`;
